components/ai_chat/core/browser/engine/engine_consumer_claude.cc

```diff
@@ -252,17 +252,12 @@ void EngineConsumerClaudeRemote::GenerateAssistantResponse(
     const std::string& human_input,
     GenerationDataCallback data_received_callback,
     GenerationCompletedCallback completed_callback) {
-  if (conversation_history.empty()) {
-    std::move(completed_callback).Run(base::unexpected(mojom::APIError::None));
-    return;
-  }
-
-  const mojom::ConversationTurnPtr& last_turn = conversation_history.back();
-  if (last_turn->character_type != CharacterType::HUMAN) {
+  if (ShouldDoEarlyCompletion(conversation_history)) {
     std::move(completed_callback).Run(base::unexpected(mojom::APIError::None));
     return;
   }
 
+  const auto& last_turn = conversation_history.back();
```

Comment:
Now I'm just confused why we have any checks here at all - from what I can tell, EngineConsumer::GenerateAssistantResponse is only called after history is added. So is this a sanity check?

Reply:
I'm not sure why it was there in the first place. I just moved the same checks across different models into a share function and then added exclusion for page refine event otherwise I won't get any responses. Maybe these checks are not needed after https://github.com/brave/brave-core/pull/24921

Reply:
In EngineConsumer's perspective, it can't see how the function is being called so yes it is a sanity check and I added missing unit test case for the early return cases.

Reply:
Perhaps it should be `bool CanPerformCompletionRequest()` then

Reply:
Sounds better and clearer, I will change the name.

Reply:
name updated